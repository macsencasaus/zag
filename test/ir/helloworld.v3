// RUN: %vector -emit-ir %s

extern fn puts(str: *u8) i32;

fn main() i32 {
    puts(&"Hello, world!");
    puts(&"Bruh moment");
    return 0;
}
