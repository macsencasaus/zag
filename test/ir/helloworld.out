; V3 Readable Intermediate Representation
; Function header formatted as:
; @func[<max-stack-size>](<param-type>,...)

extern @puts(ptr) i32

@main[16]() i32:
    %s[8] = ref i8 %data[0]
    %s[0] = i32 call @puts(ptr %s[8])
    %s[8] = ref i8 %data[14]
    %s[0] = i32 call @puts(ptr %s[8])
    ret i32 0


Data:
  0: 48 65 6C 6C 6F 2C 20 77 6F 72 6C 64          Hello,.world
  c: 21  0 42 72 75 68 20 6D 6F 6D 65 6E          !.Bruh.momen
 18: 74  0                                        t.
